### PR TITLE
chore(binding-http): enable eslint/strict-boolean-expressions

### PR DESCRIPTION
**Pull request by JKRhb**
Ticks one of the boxes of #1046 :)

### CODE REVIEW

File: packages/binding-http/src/oauth-token-validation.ts
Comment (danielpeintner) [inline packages/binding-http/src/oauth-token-validation.ts:64]: Somehow the error message does not seem right.. or confuses my.. the code checks whether both are null.. but the error says just one method is allowed?

Comment (JKRhb) [reply packages/binding-http/src/oauth-token-validation.ts:64]: Oh, good point! Seems like the logic was intended to check for the exact opposite, right (i.e., `headerToken != null && queryToken != null`)?

Comment (danielpeintner) [reply packages/binding-http/src/oauth-token-validation.ts:64]: I think so too but let's wait for @relu91 to comment...

**Comment by JKRhb** [comment]
(Rebased the PR against the latest state of `master`.)

File: packages/binding-http/src/oauth-token-validation.ts
Comment (relu91) [reply packages/binding-http/src/oauth-token-validation.ts:64]: yes, exactly the translation is wrong. It should be as [suggested](https://github.com/eclipse-thingweb/node-wot/pull/1097/files#r1335868880).

Comment (JKRhb) [reply packages/binding-http/src/oauth-token-validation.ts]: Since it is more than a refactoring but a change of the actual logic, should we maybe make this change via a separate PR? Or is fine to keep it in here?

Comment (danielpeintner) [reply packages/binding-http/src/oauth-token-validation.ts]: I think in this case I am fine with integrating it as part of this PR also. But either way is fine..

Comment (JKRhb) [reply packages/binding-http/src/oauth-token-validation.ts]: Okay, then we should be ready to merge this one :)

File: packages/binding-http/src/codecs/tuya-codec.ts
Comment (relu91) [inline packages/binding-http/src/codecs/tuya-codec.ts:35]: is this allowed?
```suggestion
        if (!(parsedBody.success ?? false)) {
```

[source](https://typescript-eslint.io/rules/strict-boolean-expressions/): See last correct example

File: packages/binding-http/src/credential.ts
Comment (relu91) [inline packages/binding-http/src/credential.ts:217]: See above

File: packages/binding-http/src/routes/property.ts
Comment (relu91) [inline packages/binding-http/src/routes/property.ts:109]: See above

File: packages/binding-http/src/http-server.ts
Comment (relu91) [inline packages/binding-http/src/http-server.ts:258]: Noticed that this getPort method is different from the http-client-basic-test, can you explain why?

File: packages/binding-http/src/http-server.ts
Comment (relu91) [inline packages/binding-http/src/http-server.ts:320]: Should we just disable the rule? or is it maybe that endsWith can result in undefined? 🤔

File: packages/binding-http/src/http-server.ts
Comment (relu91) [inline packages/binding-http/src/http-server.ts:337]: See above

File: packages/binding-http/src/http-server.ts
Comment (relu91) [inline packages/binding-http/src/http-server.ts:340]: See above

File: packages/binding-http/src/http-server.ts
Comment (relu91) [inline packages/binding-http/src/http-server.ts:378]: See above
```suggestion
                if (thing.properties[propertyName].readOnly ?? false) {
```

File: packages/binding-http/src/http-server.ts
Comment (relu91) [inline packages/binding-http/src/http-server.ts:384]: See above
```suggestion
                } else if (thing.properties[propertyName].writeOnly ?? false) {
```

File: packages/binding-http/src/http-server.ts
Comment (relu91) [inline packages/binding-http/src/http-server.ts:568]: Given the comment, this should probably be: `if(thin.security == null)`. Right?

**Review by relu91 — CHANGES_REQUESTED** [review]
I selected request changes but they are mostly suggestions and comments. I'm open to changing my mind on the suggested changes, let me know!

File: packages/binding-http/src/http-server.ts
Comment (JKRhb) [reply packages/binding-http/src/http-server.ts]: Very good point, I copied over the code from there to here :)

In general, though, these getters should be revisited later IMHO as returning `-1` does not seem like a very good practice to me. Maybe that is why I forgot to also provide a "proper" fix here :/

Another thing I noticed, by the way, is that we should consider using getters and setters as defined by JavaScript for this kind of method in the future (see, e.g., [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) for an example) which is more idiomatic and simplifies some of the APIs.

The method implementation would then become something like this:

```ts
    class HttpServer {
        public get port(): number {
            return this.#port;
        }
    }
```

While the respective properties could be accessed like so (effectively getting rid of the `get` prefix):

```ts
    const serverPort = httpServer.port;
```

File: packages/binding-http/src/http-server.ts
Comment (JKRhb) [reply packages/binding-http/src/http-server.ts:568]: As `thing.security` is a mandatory field, I think this part of the code covers the `thing.security.length === 0` case (i.e., the test for `thing.security.length > 0` above does not return true and we did not return from the method early).

This could definitely be a bit clearer, though – maybe we can perform some additional cleanups after this PR has been merged?

File: packages/binding-http/src/codecs/tuya-codec.ts
Comment (JKRhb) [reply packages/binding-http/src/codecs/tuya-codec.ts:35]: I think both versions are fine here – in the end, it is probably a question of readability 🤔 Due to otherwise doubled negation, I actually tend to the `!== true` version in this instance but I would also be fine with nullish coalescing :)

Comment (JKRhb) [reply packages/binding-http/src/codecs/tuya-codec.ts]: In order to maximize readability, I now introduced an `success` auxiliary variable – so I think, this should now be resolved :)

File: packages/binding-http/src/http-server.ts
Comment (JKRhb) [reply packages/binding-http/src/http-server.ts:324]: I think this issue as well as others should get resolved by #1103, I will therefore remove the `=== true` check as well as the TODO :)

**Comment by JKRhb** [comment]
Hmm, the persisting linting errors are still puzzling me a bit, as I don't get the same errors on my own machine. I think I now remember, though, that they were the reason why I used the `=== true` approach in the first place. It seems to me, though, that some of the errors are caused by the TS config and using `ES6` as the compile target.

I think the easiest approach would be to stick to `=== true` for now and maybe revisit this issue later.

**Comment by JKRhb** [comment]
Since the updated eslint packages did not solve the issue and to make the history a bit cleaner, I dropped the commits merged in from #1103 and rebased against `master`.

File: packages/binding-http/src/http-server.ts
Comment (relu91) [reply packages/binding-http/src/http-server.ts]: +1 for both considerations. However, I would address those in a separate PR. Regarding this PR we can consider this issue resolved.

File: packages/binding-http/src/http-server.ts
Comment (relu91) [reply packages/binding-http/src/http-server.ts:568]: It is a mandatory field in the TD but not in the `ThingExposeInit`. In my understanding, there is nothing so far to check if the `security` is defined. Probably, due to #911. Currently, we just ask the bindings to handle the `ExposeThingInit` see https://github.com/eclipse-thingweb/node-wot/blob/master/packages/core/src/servient.ts#L38 .  So `thing.security` might be also `null` at that point in time.   Do the new changes take care of this too? 🤔 .

Comment (JKRhb) [reply packages/binding-http/src/http-server.ts:568]: Ohh, I see, then this case might not be covered yet! I will double-check this :)

Comment (JKRhb) [reply packages/binding-http/src/http-server.ts:568]: Hmm, the parameter passed into the method is of type `security: string | [string, ...string[]]` – so it is mandatory, but I suppose at least one thing that is missing is the case when security is a single string?

File: packages/binding-http/src/http-server.ts
Comment (JKRhb) [reply packages/binding-http/src/http-server.ts]: I now managed to get rid of the `=== true` approach using some explicit typing :)

File: packages/binding-http/src/http-server.ts
Comment (JKRhb) [reply packages/binding-http/src/http-server.ts:568]: I tried to provide a fix now in 7a0bd0240de661a9e2df5981038a48cfceb92ba4 – I think that should work for now, however, it seems to me as if the handling of `security` and `securityDefinitions` needs to be revisited in general.

Comment (JKRhb) [reply packages/binding-http/src/http-server.ts:568]: Aparently, the code has been correct before, so it reverted the change. Seems to me as if the typing of the `security` field in the `ExposedThing` class/interface might need to be fixed.

**Comment by JKRhb** [comment]
Rebased against `master` to resolve a conflict – with the latest changes, I think/hope this PR should be ready for merging now :)

File: packages/binding-http/src/http-server.ts
Comment (relu91) [reply packages/binding-http/src/http-server.ts:568]: I know that it is counter-intuitive and I agree that the bug is somewhere else (e.g. #911). If you go upwards in the call hierarchy you'll notice that after producing method nobody checks if `security` is defined, tomorrow I'll try with a practical example. However for the time being we can merge the PR and track this issue in #911 .

**Review by relu91 — APPROVED** [review]
All the comments above have been resolved. Merging. Thanks @JKRhb !!!

**Comment by JKRhb** [comment]
Awesome, thank you @relu91! :)

Maybe one thing, though: It's probably best to use the squashing strategy when merging PRs like this one, as the `fixup!` commits I am using are actually intended to be squashed eventually (see, e.g., [here](https://blog.sebastian-daschner.com/entries/git-commit-fixup-autosquash)). After the discussions we had recently, I refrained from doing so which now, unfortunately, led to some of those commits "leaking" into the main branch. Maybe we could discuss our workflow with handling PRs more generally at one of the upcoming committer meetings? Potential solutions could be to let the person opening the PR merge the PR after receiving the required approvals, provided that it is a committer, or let them also give a final green light before the merge.

Personally, I would prefer if we would allow to also perform some clean-up in the PR (i.e., rebasing, squashing, and reordering of commits) itself *after* the reviewers gave their approval since having a clean commit history in the main branch is always a nice thing to have, I think. But maybe we could discuss this also at some point (in not too much detail) in a synchronous manner (which would also probably lead to the resolution of #558).

**Comment by danielpeintner** [comment]
As said before.. rebasing, squashing, and reordering of commits while working on a PR makes it very difficult to follow what has changed between commits.

I guess the only thing we need to remember (when "actually" merging the PR) is set/choose the right option to "squash all commits".
I think on GitHub we can specify the default strategy ... we might ask Eclipse to do this setting... IF EVERYONE AGREES WITH ... which isn't clear from reading https://github.com/eclipse-thingweb/node-wot/issues/558

Shall we continue the discussion in https://github.com/eclipse-thingweb/node-wot/issues/558

**Comment by JKRhb** [comment]
> As said before.. rebasing, squashing, and reordering of commits while working on a PR makes it very difficult to follow what has changed between commits.

You are completely right, that should only be done once all reviews have been addressed. Sorry once more for not following that principle before.

> I guess the only thing we need to remember (when "actually" merging the PR) is set/choose the right option to "squash all commits". I think on GitHub we can specify the default strategy ... we might ask Eclipse to do this setting... IF EVERYONE AGREES WITH ... which isn't clear from reading #558
> 
> Shall we continue the discussion in #558

:+1:

**Pull request merged**
